app/models/supermarket.rb

```diff
@@ -1,3 +1,5 @@
 class Supermarket < ApplicationRecord
+  validates_presence_of :name, :location
+
   has_many :items
```

Comment:
In order to be able to call `@supermarket.customers` in your controller, you need to add this association to directly link supermarkets to customers (they are already linked via supermarket -> items -> customer_items -> items but we want to tell AR to to do this DIRECTLy! Because this is a lengthy join otherwise.

How can we set up this direct link from supermarket to customers?: Set up the Supermarket model to take advantage of the associations that items, its child object has. Similar to how you use it with many-to-many relationships, you could add the following association to the Supermarket model:
`has_many :customers, through: :items`

Under the hood, AR will build an association that gets all the customers associated with the items that belong to this store. This is a multi-step join under the hood, but with AR, we can just call `supermarket.customers` and get the results!